### ISSUE TITLE
Is It possible to add an ebuild for node-red ?

### ISSUE THREAD

**Issue by zen2** [onkelbeh/HomeAssistantRepository#18]
Since I use your overlay, I wish to have an ebuild for node-red that I use, install and update actually by hand:
https://github.com/node-red/node-red

**Comment by onkelbeh** [comment]
Could you show what you do with node-red (for my personal interest)?

**Comment by zen2** [comment]
I use simply node-red to manage all my HA automation and it permits me to avoid completely YAML for automation.

It can be used in two way:

- directly to speak with MQTT so your automations are not dependant on HA that become a simple optional dashboard

- to commuicate directly with HA via a websocket using:
   - node-red-contrib-home-assistant-websocket palette on node-red
   - node-red integration on HA

I use the second option so you got node-red nodes specifically for HA that permit to listen for events, read entity states, launch HA actions....
I've added node-red as panel_iframe on HA side so I can access it directly from HA to configure my automations.

Node-red is really popular among HA users because it permits really to create complex automations easier than within HA. With websocket support it permits even to create entities as for example binary sensors that can be discovered by HA.
The node-red UI is well designed and there is almost no bug, the project is rock solid maintain project (At origin node-red was designed to create workflow to train IBM Watson IA).
Basically It's an editor of workflow that uses nodes written in JS. A thematic collection of nodes is called a palette (Examples: MQTT palette, HA palette, ....). So you can add already designed palette to add specific nodes that you need to use. Note that a powerfull side of node-red is that you can copy/paste as JSON text easily any workflow of part of it to share it on web.

I advice you to try the quick tutorial on node-red web site (15mn) to understand principles of node-red and let you understand why it can be so powerfull when used for domotic system.

My current domotic stack is:
Zigbee / Zigbee2mqtt / node-red / HA

**Comment by RayOfLight1** [comment]
@onkelbeh I'm running node-red too, I've used the instructions found in https://nodered.org/docs/getting-started/local

I only executed `sudo npm install -g --unsafe-perm node-red`

I still have not set it up to start as a daemon at startup, which it would make it more convenient.

I'm sad that I installed it through npm and not through portage though, i still have not had to "update" it.

I can show you some (not very advanced) automations i have running that affect Home Assistant if you are interested.

**Comment by zen2** [comment]
> @onkelbeh I'm running node-red too, I've used the instructions found in https://nodered.org/docs/getting-started/local
> 
> I only executed `sudo npm install -g --unsafe-perm node-red`
> 
> I still have not set it up to start as a daemon at startup, which it would make it more convenient.
> 
> I'm sad that I installed it through npm and not through portage though, i still have not had to "update" it.
> 
> I can show you some (not very advanced) automations i have running that affect Home Assistant if you are interested.

I advice you to install node-red on a specific user path. Install globally nodejs stuff in system outside portage can bring problems later. See below.

**Create user and install node-red**
```
useradd -m -d /path/to/node-red nodered
su - nodered
npm install node-red
```

**Update node-red:**
```
su - nodered
cd .node-red
npm install node-red
```

**Openrc init.d service:**

_/etc/init.d/node-red_
```
#!/sbin/openrc-run
# Copyright 2020 Gentoo Authors
# Distributed under the terms of the GNU General Public License v2

name="${SVCNAME} daemon"
description="${DESCRIPTION}"
command_args="${OPTIONS}"

start() {
  ebegin "Starting ${name}"
    start-stop-daemon --start --quiet --background \
                --user ${USER:-nobody} \
                --group ${GROUP:-nobody} \
                --exec "${COMMAND}" \
                --pidfile "${PID:-/run/${SVCNAME}.pid}" \
                --make-pidfile
  eend $?
}

stop() {
  ebegin "Stopping ${name}"
  start-stop-daemon --stop \
                --pidfile "${PID:-/run/${SVCNAME}.pid}" \
  eend $?
}
```
_/etc/conf.d/node-red_
```
DESCRIPTION='Low-code programming for event-driven applications'
COMMAND="/home/nodered/.node-red/node_modules/.bin/node-red-pi"
USER="nodered"
GROUP="nodered"
#PID=
#OPTIONS=
rc_need="mosquitto"
```

**Comment by inode64** [comment]
I make a pull with the ebuild for this in https://git.edevau.net/onkelbeh/HomeAssistantRepository/pulls/1090

**Comment by onkelbeh** [comment]
Thanks @inode64:
https://github.com/onkelbeh/HomeAssistantRepository/commit/db5cba5d7ce381712552b8b1248449c18d70ccd6
https://github.com/onkelbeh/HomeAssistantRepository/tree/master/app-misc/node-red
